### ISSUE TITLE
BgQuery Conflict between

### ISSUE THREAD

**Issue by max-ostapenko** [sqlfluff/sqlfluff#5893]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

CTE closing bracket is on the same line and `layout.cte_bracket` is excluded.
Linting still raises an error for `layout.indent`.

```sql
WITH zoo AS (
    SELECT a FROM foo)

SELECT * FROM zoo
```

```sh
$ sqlfluff lint qwe.sql
== [example.sql] FAIL                                                                                                         
L:   2 | P:  22 | LT02 | Expected line break and no indent before ')'.
                       | [layout.indent]
All Finished 📜 🎉!
```

and expects the following:
```sql
WITH zoo AS (
    SELECT a FROM foo
)

SELECT * FROM zoo
```

### Expected Behaviour

Excluding `layout.cte_bracket` rule allows to have the following format as expected:
```sql
WITH zoo AS (
    SELECT a FROM foo)

SELECT * FROM zoo
```

### Observed Behaviour

Despite that this query and expected behaviour [explicitly documented for LT07](https://github.com/sqlfluff/sqlfluff/blob/dd04c83c9e9ae20eb28a635fc0503f2dc002ab56/src/sqlfluff/rules/layout/LT07.py#L12) there is some conflict with `layout.indent`.

Would be helpful to clarify the interplay between these rules and have an option to configure it.

### How to reproduce

1. Create `example.sql`:
```sql
WITH zoo AS (
    SELECT a FROM foo)

SELECT * FROM zoo
```

2. Create `.sqlfluff`:
```
[sqlfluff]
dialect = bigquery
exclude_rules = layout.cte_bracket
```

3. Run linting:
```sh
$ sqlfluff lint qwe.sql
```

### Dialect

bigquery

### Version

sqlfluff, version 3.0.6
Python 3.12.2 (main, Feb  6 2024, 20:19:44) [Clang 15.0.0 (clang-1500.1.0.2.5)] on darwin

### Configuration

[sqlfluff]
dialect = bigquery
exclude_rules = layout.cte_bracket

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
